app/controllers/users_controller.rb

```diff
@@ -0,0 +1,55 @@
+class UsersController < ApplicationController
+  skip_before_action :require_login, only: [:new, :create]
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      login(params[:user][:email], params[:user][:password])
+      redirect_to root_path
+    else
+      render "new"
+    end
+  end
+
+  def settings
+    render template: "users/settings"
+  end
+
+  def edit
+  end
+
+  def update
+    password     = params[:user][:password]
+    new_password = params[:user][:new_password]
+    email        = params[:user][:email]
+
+    if User.authenticate(current_user.email, password) || current_user.external?
+      if email
+        current_user.password = password
+        current_user.email = email
+        current_user.save
+        flash[:success] = 'Email was successfully updated.'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.